src/hypothesis/strategies.py

```diff
@@ -695,6 +695,32 @@ def text(
 
 @cacheable
 @defines_strategy
+def strings_matching_regex(regex):
+    """Return strategy that generates strings that match given regex.
+
+    Regex can be either a string or compiled regex (through
+    :func:`re.compile`).
+
+    You can use regex flags (such as :const:`python:re.IGNORECASE`,
+    :const:`python:re.DOTALL` or :const:`python:re.UNICODE`) to control
+    generation. Flags can be passed either in compiled regex (specify flags in
+    call to :func:`re.compile`) or inside pattern with (?iLmsux) group.
```

Comment:
As above, use the `python:re.compile` namespacing so config changes break this rather than linking somewhere unexpected.